### PR TITLE
feat(angular-material): upgrade to 16

### PR DESCRIPTION
**Pull request by romdhanisam**
Fixes: #7800

> BREAKING CHANGE: ⛵upgrade angular-material to 16

### CODE REVIEW

File: modules/web/package.json [hunk 1]
```diff
@@ -44,14 +44,14 @@
   },
   "dependencies": {
     "@angular/animations": "16.2.1",
-    "@angular/cdk": "16.2.1",
+    "@angular/cdk": "~16.2.1",
```
Comment (shu-mutou) [inline modules/web/package.json:47]: Could you specify version strictly? Please remove `~`.

File: modules/web/package.json [hunk 2]
```diff
@@ -44,14 +44,14 @@
...
     "@angular/common": "16.2.1",
     "@angular/compiler": "16.2.1",
     "@angular/core": "16.2.1",
     "@angular/flex-layout": "15.0.0-beta.42",
     "@angular/forms": "16.2.1",
     "@angular/localize": "16.2.1",
-    "@angular/material": "14.2.2",
+    "@angular/material": "~16.2.1",
```
Comment (shu-mutou) [inline modules/web/package.json:54]: ditto

**Comment by romdhanisam** [comment]
Same PR here https://github.com/kubernetes/dashboard/pull/8345

**Comment by romdhanisam** [comment]
> Upgrading material will require changing our whole theming system since it does not work properly with the newest version.

Hi, I've made all the needed changes here https://github.com/kubernetes/dashboard/pull/8345

File: modules/web/package.json (hunk 1 above)
Comment (romdhanisam) [reply modules/web/package.json]: Hi, I've made all the needed changes here https://github.com/kubernetes/dashboard/pull/8345

**Pull request closed without merging**